src/core.jl

```diff
@@ -19,3 +19,187 @@ immutable ImageEdge
     index2::Int
     weight::Float64
 end
+
+"""
+    G, vert_map = region_adjacency_graph(seg, weight_fn)
+
+Constructs a region adjacency graph from the `SegmentedImage`. It returns the RAG
+along with a label->vertex map. `weight_fn` is used to assign weights to edges.
```

Comment:
Also, API-wise would it be better to make sure that the vertices are numbered in the same way as `segment_labels`? How much of an advantage do we get from renumbering them? I recognize that we might have some disconnected nodes in the graph, but how problematic is that?